### DISCUSSION TITLE
Update speed

### DISCUSSION THREAD

**Discussion by emeybee** [ESPresense/ESPresense#504] [Support]
Hi, I'm using ESPresense to try to track my kitten around the house, in an attempt to turn off the electric dog door via Home Assistant before she dashes out of it. So far it hasn't worked because she moves too fast.

I was using a Trackr Pixel because of the small size, but she kept beating it out the door, so I thought it wasn't sending out enough updates. I tried a Tile and that was even worse. I've now gotten a Blue Charm Beacon which is configured to update every 500 milliseconds. But even though it's transmitting every half second, ESPresense is still taking a long time to update. If I move the beacon, even right up next to the ESP32, then Raw updates within about 8 seconds, but Distance takes 20+. . This is in both Home Assistant and MQTT Explorer.

Are there any settings changes I can make so ESPresense and/or Home Assistant update faster? I just went with the defaults on install except for setting up the WIFI and turning off the LED. I'm attaching screenshots of my settings:
![settings2](https://user-images.githubusercontent.com/74482237/176099163-437f4424-532e-420c-8c94-130378e69bd3.PNG)
![settings1](https://user-images.githubusercontent.com/74482237/176099167-da76fa3d-0bc7-4354-baeb-de809a6d8505.PNG)

Here is the output from MQTT Explorer:
Living room: {"id":"iBeacon:fda50693-a4e2-4fb1-afcf-c6eb07647825-5-6","name":"emeybee","idType":175,"rssi@1m":-80,"rssi":-96,"raw":2.87,"distance":1.69,"speed":0.03,"mac":"d19cedf9e67e","interval":3126}
Hallway: {"id":"iBeacon:fda50693-a4e2-4fb1-afcf-c6eb07647825-5-6","idType":175,"rssi@1m":-80,"rssi":-92,"raw":2.2,"distance":2.03,"speed":0.01,"mac":"d19cedf9e67e","interval":1218}

Also this is a dumb question but the distance is in meters, correct? Because right now I have it at least 5 meters from the hallway tracker and it's saying 2.03... though some of the distance is from going upstairs to downstairs so does it not register vertical distance?

This is the ESP32 I have if that makes a difference: https://www.amazon.com/gp/product/B07QCP2451/

Thank you!

**Comment by gnome4711** [comment]
Hi!

Set the following options:
![grafik](https://user-images.githubusercontent.com/7687563/176186053-29e81922-9d63-4bf1-ae61-19885668df69.png)

Works pretty fast for me



gnome

**Comment by emeybee** [comment]
Ok thanks! I will give it a try when I get home from work tonight :)

**Reply by karatecarter** [reply]
Did you have any luck with this?  I have had the same problem - I've been testing with two D1 Minis on my desk about 6ft apart and I put my phone right next to one then the other, and I notice that "raw" updates pretty fast but "distance" (which is what Home Assistant seems to use to determine device location) is a lot slower to update, as it appears to be the average distance seen over the last x reports.  I also kind of wonder if I have a few of these reporting to the MQTT server this often is it going to have an impact on my local network congestion?

**Reply by emeybee** [reply]
I don't have it working perfect by any means, but it's a lot better... these are my current settings. It's been a bit so I don't remember exactly which one helped... I also filtered out anything except the specific beacons I want to track.

![espresense](https://user-images.githubusercontent.com/74482237/192344658-0d0cdaa0-c1ea-4a44-932d-41cbf29b5a01.PNG)

**Reply by karatecarter** [reply]
Thanks for sharing your settings.  Have you found that some devices update a lot faster than others?  I am wondering if my problem is that I've been mainly testing with my iPhone.  I found out you can view the data from the device rather than relying on the mqtt data by navigating to http://ip/ui/#/fingerprints, so I checked that and noticed that when I brought my iPhone from a distance of 8 meters to about 1 meter, it took over 40 seconds for the distance to drop down to 1.  The "raw" value in mqtt seems to update pretty fast, but the "distance" value (which I assume is something like the average over the last x readings) is a lot slower.  I'm wondering if there is a way to configure how many readings it uses to compute the "distance".

**Reply by emeybee** [reply]
The device definitely makes a difference. I tried a Tile tracker, a Trackr Pixel and a Blue Charm Beacon. The Tile only updated every few minutes so was pretty useless for my needs. The Trackr was better but my kitten still outran it. The Blue Charm Beacon has worked pretty well and the guy who runs the company has been extremely helpful with figuring out the best settings and troubleshooting issues, even though they were on the ESPresense end and not the fault of the beacon.

I remember reading something about Apple scrambling their Bluetooth signal, or maybe changing the name or something, for privacy concerns, so that may cause you additional issues.
